### PR TITLE
Add transform method for NamedArgBindingPatternNode.

### LINKED ISSUE #25985: NamedArgBindingPatternNode not supported in transformer
**Description:**
consider the following BBE
```
import ballerina/io;

type SampleErrorData record {
    string message?;
    error cause?;
    string info;
    boolean fatal;
};

type SampleError error<string, SampleErrorData>;

public function main() {
    string reason;
    string info;
    boolean fatal;
    // This error-destructure binding pattern will destructure an `error` value of the type `SampleError`
    // and assign the values to the relevant variable references.
    // The value of the reason string in the `SampleError` will be assigned to the variable `reason`.
    // The values in the detail mapping will be assigned to the relevant variables.
    error(reason, info = info, fatal = fatal) = getSampleError();
    io:println("Reason String: " + reason);
    io:println("Info: ", info);
    io:println("Fatal: ", fatal);

    string reasonTwo;
    map<anydata|error> params;
    // The detail mapping can be destructured into an existing `map<anydata|error>`-typed variable by using a rest parameter.
    error(reasonTwo, ...params) = getSampleError();
    io:println("Reason String: ", reasonTwo);
    io:println("Detail Map: ", params);
    
    // The underscore '_' sign can be used to ignore either the reason string or the detail mapping.
    string? detailMsg;
    error(_, detailMsg = detailMsg) = getRecordConstrainedError();
    io:println("Detail Message: ", detailMsg);
}

function getSampleError() returns SampleError {
    SampleError e = error("Sample Error", info = "Detail Info", fatal = true);
    return e;
}

type Foo record {|
    string message?;
    error cause?;
    string detailMsg;
    boolean isFatal;
|};

function getRecordConstrainedError() returns error<string, Foo> {
    error<string, Foo> e = error("Some Error", detailMsg = "Failed Message", isFatal = true);
    return e;
}
```
Gives a bad sad error
```
[2020-09-21 17:07:06,803] SEVERE {b7a.log.crash} - Node not supported: NamedArgBindingPatternNode 
java.lang.RuntimeException: Node not supported: NamedArgBindingPatternNode
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transformSyntaxNode(BLangNodeTransformer.java:3601)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transformSyntaxNode(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.NodeTransformer.transform(NodeTransformer.java:624)
	at io.ballerinalang.compiler.syntax.tree.NamedArgBindingPatternNode.apply(NamedArgBindingPatternNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2348)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.ErrorBindingPatternNode.apply(ErrorBindingPatternNode.java:63)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createActionOrExpression(BLangNodeTransformer.java:4208)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createExpression(BLangNodeTransformer.java:4179)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createErrorDestructureStatement(BLangNodeTransformer.java:2461)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2426)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.AssignmentStatementNode.apply(AssignmentStatementNode.java:58)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.generateAndAddBLangStatements(BLangNodeTransformer.java:4090)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1428)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.FunctionBodyBlockNode.apply(FunctionBodyBlockNode.java:59)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createFunctionNode(BLangNodeTransformer.java:1335)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1306)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.FunctionDefinitionNode.apply(FunctionDefinitionNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:530)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:435)
	at io.ballerinalang.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:460)
	at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:126)
	at org.wso2.ballerinalang.compiler.parser.Parser.parse(Parser.java:99)
	at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:487)
	at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:374)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackage(Compiler.java:220)
	at org.wso2.ballerinalang.compiler.Compiler.compile(Compiler.java:101)
	at org.wso2.ballerinalang.compiler.Compiler.build(Compiler.java:109)
	at org.ballerinalang.packerina.task.CompileTask.execute(CompileTask.java:58)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.RunCommand.execute(RunCommand.java:297)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:57)
```

**Suggested Assignees (optional):**

- @rdulmina

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$Subject

Fixes #25985

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (dulajdilshan) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java:2]:
```suggestion
 *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.BRunUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Test cases for Error destructure binding pattern.
+ *
+ * @since 1.2
```
Comment (dulajdilshan) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java:31]: Should be 2.0 Swan lake right ?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java (hunk 1 above)
Comment (rdulmina) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java]: Updated thanx.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java (hunk 2 above)
Comment (rdulmina) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorDestructureTest.java]: Updated thanx

**Review by SupunS — APPROVED** [review]

**Pull request merged**